packages/containers/src/ComponentForm/kit/createTriggers.js

```diff
@@ -0,0 +1,185 @@
+/**
+ *  Copyright (C) 2006-2018 Talend Inc. - www.talend.com
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+/* eslint-disable no-param-reassign */
+
+import get from 'lodash/get';
+import isEqual from 'lodash/isEqual';
+import merge from 'lodash/merge';
+
+import flatten from './flatten';
+import defaultRegistry from './defaultRegistry';
+
+const DEFAULT_HEADERS = {
+	'Content-Type': 'application/json',
+	Accept: 'application/json',
+};
+
+function noOpTrigger({ error, trigger }) {
+	// eslint-disable-next-line no-console
+	console.error(`${JSON.stringify(trigger)} failed with error ${error || '-'}`);
+}
+
+/**
+ * array are describe without their index
+ * use the schema to guess the path to use
+ * to get the value in the properties
+ * @param {string} specPath the path provided by the trigger
+ * @param {Object} schema the schema of the current field
+ * @return {string} path to get the value in properties
+ */
+export function getPathWithArrayIndex(specPath, schema) {
+	if (!schema) {
+		return specPath;
+	}
+	const contextualPathItems = schema.key;
+	if (!specPath || !contextualPathItems) {
+		return specPath;
+	}
+	let keyIndex = 0;
```

Comment:
you should provide this value to the reduce function, to avoid side effects
{path: '', keyIndex:0}